src/Dex/Foreign/Context.hs

```diff
@@ -105,7 +107,7 @@ dexLookup :: Ptr Context -> CString -> IO (Ptr Atom)
 dexLookup ctxPtr namePtr = do
   Context _ env <- fromStablePtr ctxPtr
   name <- peekCString namePtr
-  case lookupSourceName env $ fromString name of
+  case lookupSourceName env (fromString name) :: Except AnyBinderInfo of
```

Comment:
Why do we need those type annotations now?

Reply:
Without them, GHC complains about ambiguous `e` in `Either e a`. Previously it could figure out the `e` from the constraint `MonadError Err`. It's annoying to have to write these annotations. One way around it would be to make `Except` a proper type in its own right, instead of an alias for `Either Errs`.

Reply:
Ok, I just made `Except` a proper type. It wasn't too bad at all and it has other benefits.